src/ucs/datastruct/string_buffer.c

```diff
@@ -68,60 +64,59 @@ void ucs_string_buffer_appendf(ucs_string_buffer_t *strb, const char *fmt, ...)
     va_list ap;
     int ret;
 
-    ucs_array_reserve(string_buffer, &strb->str,
-                      ucs_array_length(&strb->str) + UCS_STRING_BUFFER_RESERVE);
+    ucs_array_reserve(strb, ucs_array_length(strb) + UCS_STRING_BUFFER_RESERVE);
```

Comment:
what about changing the macro to simply add free space, in order to use it like `ucs_array_reserve(strb, UCS_STRING_BUFFER_RESERVE)`? seems we currently mostly use it with the added `ucs_array_length()` anyways

Reply:
There are few places where we reserve with the actual size:
https://github.com/openucx/ucx/blob/bd8433bfee918cf346578f1690cde030d16611e6/src/ucp/core/ucp_am.c#L204-L204
https://github.com/openucx/ucx/blob/bd8433bfee918cf346578f1690cde030d16611e6/src/ucp/core/ucp_worker.c#L2522-L2522

Reply:
i see